src/DS/rds.c

```diff
@@ -730,6 +732,36 @@ static int ds_rds_create_from_dom(xmlDocPtr* ret, xmlDocPtr sds_doc, xmlDocPtr x
 	xmlDOMWrapReconcileNamespaces(sds_wrap_ctxt, sds_res_node, 0);
 	xmlDOMWrapFreeCtxt(sds_wrap_ctxt);
 
+	if (tailoring_doc) {
+		const char *tailoring_component_id = "scap_org.open-scap_comp_user_file_tailoring";
+		const char *tailoring_component_ref_id = "scap_org.open-scap_cref_user_file_tailoring";
+
+		xmlDOMWrapCtxtPtr tailoring_wrap_ctxt = xmlDOMWrapNewCtxt();
+		xmlNodePtr tailoring_res_node = NULL;
+		xmlDOMWrapCloneNode(tailoring_wrap_ctxt, tailoring_doc, xmlDocGetRootElement(tailoring_doc),
+				&tailoring_res_node, doc, NULL, 1, 0);
+		xmlNsPtr sds_ns = sds_res_node->ns;
+		xmlNodePtr tailoring_component = xmlNewNode(sds_ns, BAD_CAST "component");
+		xmlSetProp(tailoring_component, BAD_CAST "id", BAD_CAST tailoring_component_id);
+		xmlAddChild(tailoring_component, tailoring_res_node);
+		xmlAddChild(sds_res_node, tailoring_component);
```

Comment:
This code aims to add a new component to a SCAP source datastream. I suggest investigating if we can reuse `ds_sds_compose_add_component` or `ds_sds_compose_add_component_with_ref` here.